File: API/Models/Pair.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Models
{
    public class Pair
    {
        public int Id {get;set;}
        public int Year {get;set;}=DateTime.Now.Year;
        public DateTime CreatedAt {get;set;}=DateTime.Now;
        public int GiverId {get;set;}
        public User Giver {get;set;}=null!;
        public int ReceiverId {get;set;}
        public User Receiver {get;set;}=null!;
    }
}

File: client/src/app/models/pair.ts
import type { User } from "./user";

export interface Pair {
  id: number;
  year: number;
  createdAt: string;
  giverId: number;
  giver: User;
  receiverId: number;
  receiver: User;
}

export interface YearPairs {
  year: number;
  pairs: Pair[];
}